core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -196,7 +208,7 @@ private boolean doEmit() {
     }
 
     private void handleCancelState() {
-        synchronized (this) {
+        synchronized (cancelLock) {
```

Comment:
So that the `upstreamSubscriber` (coming from the `upstreamResponseTransformer`, e.g. the `FileAsyncResponseTransformer.FileSubscriber`) can receive the `onComplete` signal after data has been sent.

Reply:
Ah right! Thanks for adding documentation. I'm sure I'll forget about this after two days.